skribenten-web/frontend/src/Brevredigering/LetterEditor/actions/toggleBulletList.ts

```diff
@@ -0,0 +1,436 @@
+      toggleBulletListOn(draft, literalIndex);
+    } else if (theContentTheUserIsOn.type === "ITEM_LIST") {
+      toggleBulletListOff(draft, literalIndex as ItemContentIndex);
+    }
+  },
+);
+
+/**
+ * Når vi lager et punkt, så må vi ta høyde for at det kan være en punktliste før, etter, eller ingen punktliste.
+ * Det kan finnes punktliste i blokk før, blokk etter, eller begge. Det kan også være punktliste i samme blokk, før / etter / begge, på literalen vi er på
+ *
+ * Det vil si at når vi converterer en literal til en ny punktliste, merger vi med nabo-punktlistene, hvis de eksisterer
+ * Så må vi sjekke om det finnes nabo-punktlister på blokk-nivå, og merge de også
+ *
+ * Fordi vi gjør en såpass stor endring i dokument strukturen, Så må vi oppdatere fokuset til editorstaten til å være på rett plass
+ */
+const toggleBulletListOn = (draft: Draft<LetterEditorState>, literalIndex: LiteralIndex) => {
+  //-------------- Convertering av literal til punktliste ----------------
+  const thisBlock = draft.redigertBrev.blocks[literalIndex.blockIndex];
+  const theIdexOfTheContent = literalIndex.contentIndex;
+  const sentence = getSurroundingLiteralsAndVariables(thisBlock.content, theIdexOfTheContent);
+  const sentenceElements = sentence.map((r) => r.element);
+
+  const replacedWithItemList = replaceElementsBetweenIncluding(
+    thisBlock.content,
+    sentence[0].originalIndex,
+    sentence.at(-1)!.originalIndex,
+    newItemList({
+      items: [{ id: null, content: sentenceElements }],
+    }),
+  );
+
+  const { newcontent, newDeleted } = mergeItemLists(replacedWithItemList);
+
+  const newThisParagraph: ParagraphBlock = newParagraph({
+    id: thisBlock.id,
+    content: newcontent,
+    deletedContent: [...sentenceElements.filter((s) => !!s.id).map((r) => r.id!), ...newDeleted],
+  });
+
+  draft.redigertBrev.blocks[literalIndex.blockIndex] = newThisParagraph;
+  //---------------------------------------------
+
+  //---- Merging av punktlister dersom det finnes en punktliste-blokk før ----------------
+  const doesPreviousBlockEndWithAnItemList =
+    draft.redigertBrev.blocks[literalIndex.blockIndex - 1]?.content.at(-1)?.type === "ITEM_LIST";
+
+  const doesThisBlockStartWithAnItemList = newThisParagraph.content[0].type === "ITEM_LIST";
+
+  if (doesPreviousBlockEndWithAnItemList && doesThisBlockStartWithAnItemList) {
+    const previousBlock = draft.redigertBrev.blocks[literalIndex.blockIndex - 1];
+
+    const previousBlockContentBeforeItemList = previousBlock.content.slice(0, -1);
+    const previousBlockContentItemList = previousBlock.content.at(-1) as ItemList;
+
+    const thisBlockContentItemList = newThisParagraph.content[0] as ItemList;
+    const thisBlockContentAfterItemList = newThisParagraph.content.slice(1);
+
+    const thisBlockMergedWithPreviousBlock = newParagraph({
+      id: previousBlock.id,
+      content: [
+        ...previousBlockContentBeforeItemList,
+        newItemList({
```

Comment:
Burde beholde ID fra en av itemListene. Så trenger deletedItems kun å reflektere deletedItems fra den listen som vi beholder ID for.